core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -325,7 +325,41 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments,
   override def renameEntityType(oldName: String,
                                 renameInfo: EntityTypeRename,
                                 parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] = {
+    // Extract the new entity type name from the rename info
+    val newName = renameInfo.newName
+
+    // Perform the rename in a transaction
+    val renameFuture = repository.dataSource.inTransaction { dataAccess =>
+      for {
+        // First check if the old entity type exists
+        entityTypeCount <- repository.queries.countEntities(workspaceId, oldName)
+        _ = if (entityTypeCount == 0) {
+          throw new RawlsExceptionWithErrorReport(
+            errorReport = ErrorReport(StatusCodes.NotFound, s"Can't find entity type $oldName")
+          )
+        }
+
+        // Check if the new entity type already exists
+        newTypeCount <- repository.queries.countEntities(workspaceId, newName)
+        _ = if (newTypeCount > 0) {
+          throw new RawlsExceptionWithErrorReport(
+            errorReport = ErrorReport(StatusCodes.Conflict, s"$newName already exists as an entity type")
+          )
+        }
```

Comment:
we don't have an existing subclass of `DataEntityException` that applies here - you could create one, or just throw a raw `DataEntityException`

(I realize the existing legacy code only throws `RawlsExceptionWithErrorReport`)